ApiFirst/DomainServices/Services/OrderService.cs

```diff
@@ -0,0 +1,93 @@
+﻿using DomainModels.Enums;
+using DomainModels.Models;
+using DomainServices.Interfaces;
+using EntityFrameworkCore.UnitOfWork.Interfaces;
+using Infrastructure.Data.Context;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DomainServices.Services
+{
+    public class OrderService : IOrderService
+    {
+        private readonly IRepositoryFactory _repositoryFactory;
+        private readonly IUnitOfWork _unitOfWork;
+
+        public OrderService(IUnitOfWork<WarrenContext> unitOfWork, IRepositoryFactory<WarrenContext> repository)
+        {
+            _unitOfWork = unitOfWork ??
+                throw new ArgumentNullException(nameof(unitOfWork));
+            _repositoryFactory = repository ?? (IRepositoryFactory)_unitOfWork;
+        }
+
+        public async Task<long> CreateAsync(Order model)
+        {
+            var repository = _unitOfWork.Repository<Order>();
+
+            await repository.AddAsync(model).ConfigureAwait(false);
+            await _unitOfWork.SaveChangesAsync().ConfigureAwait(false);
+            return model.Id;
+        }
+
+        public async Task<IEnumerable<Order>> GetAllAsync()
+        {
+            var repository = _repositoryFactory.Repository<Order>();
+            var query = repository.MultipleResultQuery().Include(source => source.Include(order => order.Product).Include(order => order.Portfolio));
+
+            return await repository.SearchAsync(query);
+        }
+
+        public async Task<Order> GetByIdAsync(long id)
+        {
+            var repository = _repositoryFactory.Repository<Order>();
+            var query = repository.SingleResultQuery()
+                                   .AndFilter(order => order.Id == id)
+                                   .Include(source => source.Include(order => order.Product).Include(order => order.Portfolio));
+
+            var order = await repository.FirstOrDefaultAsync(query).ConfigureAwait(false);
+
+            return order;
+        }
+
+        public async Task<IList<Order>> GetExecutableOrdersAsync()
+        {
+            var repository = _repositoryFactory.Repository<Order>();
+            var query = repository.MultipleResultQuery().AndFilter(order => order.LiquidateAt.Date <= DateTime.Now.Date);
+            var orders = await repository.SearchAsync(query);
```

Comment:
Isso vai fazer com que todas as ordens já executadas sejam executadas de novo.
```suggestion
            var repository = _repositoryFactory.Repository<Order>();
            var query = repository.MultipleResultQuery().AndFilter(order => order.LiquidateAt.Date == DateTime.Now.Date);
            var orders = await repository.SearchAsync(query);
```